python_modules/dagster/dagster/core/host_representation/external.py

```diff
@@ -486,20 +490,39 @@ def get_external_origin(self):
     def get_external_origin_id(self):
         return self.get_external_origin().get_id()
 
-    # ScheduleState that represents the state of the schedule
-    # when there is no row in the schedule DB (for example, when
-    # the schedule is first created in code)
-    def get_default_instigation_state(self):
+    @property
+    def default_status(self) -> ScheduleStatus:
+        return (
+            self._external_schedule_data.default_status
+            if self._external_schedule_data.default_status
+            else ScheduleStatus.STOPPED
+        )
+
+    def get_current_instigator_state(self, stored_state: Optional["InstigatorState"]):
         from dagster.core.scheduler.instigation import (
             InstigatorState,
             InstigatorStatus,
             ScheduleInstigatorData,
         )
 
+        if stored_state:
+            # Ignore AUTOMATICALLY_RUNNING states in the DB if the default status
+            # isn't ScheduleStatus.RUNNING - this would indicate that the schedule's
+            # default has changed
+            if (
+                stored_state.status != InstigatorStatus.AUTOMATICALLY_RUNNING
+                or self.default_status == ScheduleStatus.RUNNING
+            ):
+                return stored_state
+
         return InstigatorState(
             self.get_external_origin(),
             InstigatorType.SCHEDULE,
-            InstigatorStatus.STOPPED,
+            (
+                InstigatorStatus.AUTOMATICALLY_RUNNING
+                if self.default_status == ScheduleStatus.RUNNING
+                else InstigatorStatus.STOPPED
+            ),
```

Comment:
I had to stare at this for quite a while to wrap my head around it. Some thoughts:
* should `ScheduleStatus` be `DefaultScheduleStatus` ? Its currently very confusing the relationship with `InstigatorStatus` 
* maybe restructuring the condition checks here - maybe a `self.default_status == ScheduleStatus.RUNNING` block. Hoping to trade increased verbosity for increased clarity.